lib/tapioca/executor.rb

```diff
@@ -25,8 +25,14 @@ def initialize(queue, number_of_workers: nil)
       @items_per_worker = T.let((queue.length.to_f / @number_of_workers).ceil, Integer)
     end
 
-    sig { params(block: T.proc.params(item: T.untyped).void).void }
+    sig do
+      type_parameters(:T).params(
+        block: T.proc.params(item: T.untyped).returns(T.type_parameter(:T))
+      ).returns(T.nilable(T::Array[T.type_parameter(:T)]))
+    end
     def run_in_parallel(&block)
+      read, write = IO.pipe
+
       # If we only have one worker selected, it's not worth forking, just run sequentially
       if @number_of_workers == 1
         block.call(@queue.shift) until @queue.empty?
```

Comment:
Lol, this code path should also be returning the result of block calls. I guess this is why the other tests are failing on CI.